lib/servo.js

```diff
@@ -1,29 +1,44 @@
-/**
- * Debug dependencies
- */
-const debug = require('debug')('RRS:Nervi:Rotary');
+const { Gpio } = require('pigpio');
+
+const INPUT_RANGE = 2;
 
 /**
- * Dependencies
+ * Class representing a single servo.
+ *
+ * @class
  */
-const { Gpio } = require('pigpio');
+class Servo {
+  /**
+   * Create a Servo object.
+   *
+   * @constructor
+   * @param {Number} pin - The GPIO pin on the Raspberry Pi.
+   * @param {Number} min - Optional minimum range; default value is 1000.
+   * @param {Number} max - Optional maximum range; default value is 2500.
+   */
+  constructor(pin, min = 1000, max = 2500) {
+    this.servoPin = new Gpio(pin, { mode: Gpio.OUTPUT });
 
-const motor = new Gpio(10, {
-  mode: Gpio.OUTPUT,
-});
+    this.servoMin = min;
+    this.servoMax = max;
 
-let pulseWidth = 1000;
-let increment = 100;
+    this.pulseWidth = 1000;
+    this.increment = 100;
+  }
 
-debug('Turning on servo');
+  /**
+   * Move servo to the specified position.
+   *
+   * @function
+   * @param {Number} posArg - Target position.
+   */
+  moveTo(posArg) {
+    let pos = posArg;
 
-setInterval(() => {
-  motor.servoWrite(pulseWidth);
+    pos += INPUT_RANGE / 2;
 
-  pulseWidth += increment;
-  if (pulseWidth >= 2000) {
-    increment = -100;
-  } else if (pulseWidth <= 1000) {
-    increment = 100;
+    this.servoPin.servoWrite((pos * (this.servoMax - this.servoMin) / INPUT_RANGE) + this.servoMin);
```

Comment:
Do the calculation outside the parameter filed of `servoWrite` and store it value in a constant.